x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/analysis/analyzer/Analyzer.java

```diff
@@ -455,6 +456,10 @@ else if (plan instanceof OrderBy) {
                 else if (q.resolved() == false) {
                     return singletonList(q);
                 }
+                // qualifier resolves to a non-struct field and cannot be expanded
+                else if (DataTypes.isPrimitive(q.dataType())) {
+                    return null;
```

Comment:
Use `emptyList()` instead of `null` - allows iteration without blowing up and it's still fast as it avoid any allocation.

Reply:
I think you could simply return `singleton(q)` and avoid any special/null handling.